service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/utils/DateFormatUtil.java

```diff
@@ -1,26 +1,73 @@
 package uk.nhs.adaptors.gp2gp.ehr.utils;
 
+import static ca.uhn.fhir.model.api.TemporalPrecisionEnum.DAY;
+import static ca.uhn.fhir.model.api.TemporalPrecisionEnum.MINUTE;
+import static ca.uhn.fhir.model.api.TemporalPrecisionEnum.MONTH;
+import static ca.uhn.fhir.model.api.TemporalPrecisionEnum.YEAR;
+
+import ca.uhn.fhir.model.api.TemporalPrecisionEnum;
+
+import java.time.LocalDateTime;
 import java.time.ZoneId;
 import java.time.format.DateTimeFormatter;
-import java.time.format.DateTimeFormatterBuilder;
-import java.util.Date;
+import java.util.Optional;
+
+import org.hl7.fhir.dstu3.model.DateTimeType;
+import org.hl7.fhir.dstu3.model.DateType;
+import org.hl7.fhir.dstu3.model.InstantType;
+
+import com.google.common.collect.ImmutableMap;
 
 import uk.nhs.adaptors.gp2gp.ehr.exception.EhrMapperException;
 
 public class DateFormatUtil {
```

Comment:
We should have unit tests for this to cover various boundary and edge cases. This will be difficult to verify otherwise. A few ideas:

* Dates/times in both BST and GMT
* Differing time zone offsets in the FHIR/JSON representation